### PR TITLE
symbols: Build in Docker

### PR DESCRIPTION
**Pull request by chrismwendt**
Build `symbols` in Docker rather than outside of Docker then copying it in.

This eliminates the need for cross compilation and the resulting problems.

I think this will fix the broken #33689 (which was reverted in https://github.com/sourcegraph/sourcegraph/pull/34316).

- Some tree-sitter grammars need `g++`
- There is no `musl-g++`
- `symbols` used to be cross-compiled outside of Docker then copied into the image. That necessitated a musl compiler.

More details in [Slack](https://sourcegraph.slack.com/archives/C02FLQDD3TQ/p1650592129394729).

## Test plan

main-dry-run: https://buildkite.com/sourcegraph/sourcegraph/builds?branch=main-dry-run%2Fsymbols-build-in-docker

Ran locally, container at least starts.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,5 +1,7 @@
 .git/
 
+**/node_modules
```
Comment (efritz) [inline .dockerignore:3]: but what if we want _some_ node modules?

Comment (chrismwendt) [reply .dockerignore]: Dropped the wildcard `**/node_modules`

Comment (efritz) [reply .dockerignore]: lol was a joke

**Comment by chrismwendt** [comment]
Is this good to go @efritz? I'm still working on getting CI green.

**Comment by efritz** [comment]
@chrismwendt This breaks my local stack with:

```
------
failed to compute cache key: failed to walk /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: lstat /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: no such file or directory: exit status 1:
Building ctags docker image
#1 [internal] load build definition from Dockerfile
#1 sha256:1f5f38c0f8f0ba942e10851c561b84f30e72b6088ff5c3d1ca01e537e8271519
#1 transferring dockerfile: 2.18kB 0.0s done
#1 DONE 0.6s

#2 [internal] load .dockerignore
#2 sha256:357c16afd3cbad950d3bea43e2ece968d55f8aa91fdda0ebf92e05527d4ddcfa
#2 transferring context: 2B 0.1s done
#2 DONE 0.6s

#3 [internal] load metadata for docker.io/sourcegraph/alpine-3.12:142406_2022-04-14_8836ac3499f4@sha256:4681a48d1fb9a73fef1b540c08b3411f797351bbeda749f5dca21213a1e71526
#3 sha256:cad1f20893c94663b305c87088ac442abce7f5372781824c942b8af610ad0eda
#3 DONE 0.2s

#4 [ctags 1/3] FROM docker.io/sourcegraph/alpine-3.12:142406_2022-04-14_8836ac3499f4@sha256:4681a48d1fb9a73fef1b540c08b3411f797351bbeda749f5dca21213a1e71526
#4 sha256:a0da68b533566469b1cf1be99e07968d825f3a2d8848af57c7a77e3ca013ad29
#4 DONE 0.0s

#5 [internal] load build context
#5 sha256:1f5e6801501312130b7717f02cc60d66d90e7d6e1e73009e50f7f04df0abbb9e
#5 transferring context: 2B done
#5 DONE 0.0s

#6 [ctags 2/3] COPY cmd/symbols/ctags-install-alpine.sh /ctags-install-alpine.sh
#6 sha256:0bc8182b9a57bee8caa82b83ee7c18618092f12fed166128899069c584d94c1c
#6 ERROR: failed to walk /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: lstat /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: no such file or directory
------
 > [ctags 2/3] COPY cmd/symbols/ctags-install-alpine.sh /ctags-install-alpine.sh:
------
failed to compute cache key: failed to walk /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: lstat /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: no such file or directory
```

**Comment by efritz** [comment]
@chrismwendt This breaks my local stack with:

```
------
failed to compute cache key: failed to walk /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: lstat /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: no such file or directory: exit status 1:
Building ctags docker image
#1 [internal] load build definition from Dockerfile
#1 sha256:1f5f38c0f8f0ba942e10851c561b84f30e72b6088ff5c3d1ca01e537e8271519
#1 transferring dockerfile: 2.18kB 0.0s done
#1 DONE 0.6s

#2 [internal] load .dockerignore
#2 sha256:357c16afd3cbad950d3bea43e2ece968d55f8aa91fdda0ebf92e05527d4ddcfa
#2 transferring context: 2B 0.1s done
#2 DONE 0.6s

#3 [internal] load metadata for docker.io/sourcegraph/alpine-3.12:142406_2022-04-14_8836ac3499f4@sha256:4681a48d1fb9a73fef1b540c08b3411f797351bbeda749f5dca21213a1e71526
#3 sha256:cad1f20893c94663b305c87088ac442abce7f5372781824c942b8af610ad0eda
#3 DONE 0.2s

#4 [ctags 1/3] FROM docker.io/sourcegraph/alpine-3.12:142406_2022-04-14_8836ac3499f4@sha256:4681a48d1fb9a73fef1b540c08b3411f797351bbeda749f5dca21213a1e71526
#4 sha256:a0da68b533566469b1cf1be99e07968d825f3a2d8848af57c7a77e3ca013ad29
#4 DONE 0.0s

#5 [internal] load build context
#5 sha256:1f5e6801501312130b7717f02cc60d66d90e7d6e1e73009e50f7f04df0abbb9e
#5 transferring context: 2B done
#5 DONE 0.0s

#6 [ctags 2/3] COPY cmd/symbols/ctags-install-alpine.sh /ctags-install-alpine.sh
#6 sha256:0bc8182b9a57bee8caa82b83ee7c18618092f12fed166128899069c584d94c1c
#6 ERROR: failed to walk /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: lstat /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: no such file or directory
------
 > [ctags 2/3] COPY cmd/symbols/ctags-install-alpine.sh /ctags-install-alpine.sh:
------
failed to compute cache key: failed to walk /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: lstat /var/lib/docker/tmp/buildkit-mount198459844/cmd/symbols: no such file or directory
```

**Review by efritz — APPROVED** [review]

**Pull request merged**